lib/model.js

```diff
@@ -360,15 +361,64 @@ function _createSaveOptions(doc, options) {
   return saveOptions;
 }
 
+/**
+ * Runs validation before save if `validateBeforeSave` option is enabled.
+ * This runs regardless of the `middleware` option to ensure data integrity.
+ *
+ * @param {Object} options - Save options
+ * @param {Boolean} [options.validateBeforeSave] - If set, overrides schema-level validateBeforeSave
+ * @param {Boolean} [options.validateModifiedOnly] - If true, only validate modified paths
+ * @returns {Promise<void>}
+ * @memberOf Model
+ * @instance
+ * @method $__validateBeforeSave
+ * @api private
+ */
+
+Model.prototype.$__validateBeforeSave = async function $__validateBeforeSave(options) {
+  const hasValidateBeforeSaveOption = options &&
+      (typeof options === 'object') &&
+      ('validateBeforeSave' in options);
+
+  let shouldValidate;
+  if (hasValidateBeforeSaveOption) {
+    shouldValidate = !!options.validateBeforeSave;
+  } else {
+    shouldValidate = this.$__schema.options.validateBeforeSave;
+  }
+
+  if (shouldValidate) {
+    const hasValidateModifiedOnlyOption = options &&
```

Comment:
This would be a good candidate for optional chaining. Not a blocker but nice to fix if you have a minute